inference-engine/thirdparty/clDNN/runtime/engine.cpp

```diff
@@ -119,23 +119,40 @@ memory_ptr engine::share_surface(const layout& layout, shared_surface surf, uint
 }
 #endif  // _WIN32
 
-uint64_t engine::get_max_used_device_memory() const {
-    return peak_memory_usage.load();
+uint64_t engine::get_max_used_device_memory(allocation_type type) const {
+    uint64_t peak_memory_usage{};
+    auto iter = peak_memory_usage_map.find(type);
+    if (iter != peak_memory_usage_map.end()) {
+        peak_memory_usage = iter->second.load();
+    }
+    return peak_memory_usage;
 }
 
-uint64_t engine::get_used_device_memory() const {
-    return memory_usage.load();
+uint64_t engine::get_used_device_memory(allocation_type type) const {
+    uint64_t memory_usage{};
```

Comment:
why not concretely set this as 0?

Reply:
Empty brace also means zero-initialization. but I've added it for now.